server/graphql/v2/object/Host.js

```diff
@@ -437,6 +472,218 @@ export const Host = new GraphQLObjectType({
           };
         },
       },
+      contributionStats: {
+        type: new GraphQLNonNull(ContributionStats),
+        args: {
+          account: {
+            type: new GraphQLList(new GraphQLNonNull(AccountReferenceInput)),
+            description: 'A collection of accounts for which the contribution stats should be returned.',
+          },
+          dateFrom: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics beginning from this date.',
+          },
+          dateTo: {
+            type: GraphQLDateTime,
+            description: 'Calculate contribution statistics until this date.',
+          },
+          timeUnit: {
+            type: new GraphQLNonNull(TimeUnit),
+            defaultValue: 'YEAR',
+            description: 'The time unit of the time series',
+          },
+        },
+        async resolve(host, args, req) {
+          if (!req.remoteUser?.isAdmin(host.id)) {
+            throw new Unauthorized('You need to be logged in as an admin of the host to see the contribution stats.');
+          }
+          const where = {
+            HostCollectiveId: host.id,
+            kind: TransactionKind.CONTRIBUTION,
+            type: TransactionTypes.CREDIT,
+          };
+          const numberOfDays = getNumberOfDays(args.dateFrom, args.dateTo, host);
+          const dateRange = getFilterDateRange(args.dateFrom, args.dateTo);
+          if (dateRange) {
+            where.createdAt = dateRange;
+          }
+          let collectiveIds;
+          if (args.account) {
+            const collectives = await fetchAccountsWithReferences(args.account, { throwIfMissing: true });
+            collectiveIds = collectives.map(collective => collective.id);
+            where.CollectiveId = { [Op.in]: collectiveIds };
+          }
+          const contributionsCount = await models.Transaction.count({
+            where,
+          });
+          let contributionAmountOverTime;
+          if (args.timeUnit) {
+            const dateFrom = args.dateFrom ? moment(args.dateFrom).toISOString() : undefined;
+            const dateTo = args.dateTo ? moment(args.dateTo).toISOString() : undefined;
+            contributionAmountOverTime = await sequelize.query(
+              `SELECT DATE_TRUNC('${args.timeUnit}', "createdAt") AS "date", sum(amount) as "amount", "currency"
```

Comment:
I've actually taken this shortcut myself with the other timeseries since timeUnit is a safe enum, but we try to never use template strings in queries because that's an open door for SQL injections. We instead use `replacements` or `bind`:

```suggestion
              `SELECT DATE_TRUNC(:timeUnit, "createdAt") AS "date", sum(amount) as "amount", "currency"
```

```es6
replacements: { timeUnit }
```

I'll take care of updating the ones in `host-metrics`.

Reply:
Thanks for pointing it out. I've just refactored the two queries here to make use of `replacements`.